src/main/java/org/broadinstitute/hellbender/utils/runtime/AsynchronousStreamWriterService.java

```diff
@@ -0,0 +1,122 @@
+        previousBatch = null;
+    }
+
+    /**
+     * Request that a batch of items be written to the stream on a background thread. Any previously requested batch
+     * must have already been completed and retrieved via {@link #waitForPreviousBatchCompletion}.
+     *
+     * @param batchList a list of items to be written
+     * @param batchSize number of items to write
+     * @return {@code Future} representing this batch
+     */
+    public Future<Integer> startAsynchronousBatchWrite(final List<T> batchList, int batchSize) {
+        Utils.nonNull(batchList);
+        Utils.nonEmpty(batchList);
+
+        if (previousBatch != null) {
+            throw new IllegalStateException("Previous batch not yet complete");
+        }
+
+        previousBatch = executorService.submit(() -> {
+            try {
+                for (int i = 0; i < batchSize; i++) {
+                    T element = batchList.get(i);
+                    streamWriter.write(element.toString().getBytes());
+                }
+                // this can block, waiting for the stream to be consumed if its buffered
+                streamWriter.flush();
+                return batchSize; // return the number of items this batch was asked to write
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+        return previousBatch;
+    }
+
+    /**
+     * Waits for a batch that was previously initiated via {@link #startAsynchronousBatchWrite(List, int)}}
+     * to complete, flushes the target stream and returns the corresponding completed Future. The Future representing
+     * a given batch can only be obtained via this method once. If no work is outstanding, and/or the previous batch
+     * has already been retrieved, null is returned.
+     * @param timeout maximum time to wait for a response
+     * @param timeoutUnit units of {@code timeout}
+     * @return returns null if no previous work to complete, otherwise a completed Future
+     */
+    public Future<Integer> waitForPreviousBatchCompletion(long timeout, TimeUnit timeoutUnit) {
+        final Future<Integer> lastCompleteBatch = previousBatch;
+        if (previousBatch != null) {
+            try {
+                try {
+                    previousBatch.get(timeout, timeoutUnit);
+                } catch (ExecutionException | InterruptedException e) {
+                    throw new GATKException("Interrupted during background stream write");
+                } catch (TimeoutException e) {
+                    throw new GATKException("Timeout waiting for background stream write to complete");
+                }
+                streamWriter.flush();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            previousBatch = null;
+        }
+        return lastCompleteBatch;
+    }
+
+    /**
+     * Terminate the async writer, cancelling any outstanding work.
+     * @return true
```

Comment:
The description of return is wrong, although I don't actually know why you care about the return value here or what the caller is supposed to do with the information.

Reply:
It just tells the caller if an outstanding batch existed and cancelling it failed. javadoc updated.